[enterprise-4.10] RHDEVDOCS-5271 - Logging 5.6.6 release notes to 4.10 (#60272)
openshift/openshift-docs: modules/logging-rn-5.6.6.adoc

//module included in logging-release-notes.adoc
:_content-type: REFERENCE

> [id="logging-release-notes-5-6-6_{context}"]

Reviewer: ```suggestion
[id="logging-rn-5-6-6_{context}"]
```

= Logging 5.6.6
This release includes link:https://access.redhat.com/errata/RHBA-2023:3194[OpenShift Logging Bug Fix Release 5.6.6].

[id="logging-5-6-6-bug-fixes_{context}"]
== Bug fixes
* Before this update, dropping of messages occurred when configuring the `ClusterLogForwarder` custom resource to write to a Kafka output topic that matched a key in the payload due to an error. With this update, the issue is resolved by prefixing Fluentd's buffer name with an underscore. (link:https://issues.redhat.com/browse/LOG-3458[LOG-3458])

> * Before this update, premature closure of watches occurred in fluentd when inodes were reused and there were multiple entries with the same inode. With this update, the issue of premature closure of watches in the fluentd position file is resolved. (link:https://issues.redhat.com/browse/LOG-3629[LOG-3629])

Reviewer: Fluentd vs fluentd capitalization is inconsistent, I'd pick one and use it consistently in the doc

> * Before this update, the detection of Java Script client multi-line exceptions by fluentd failed, resulting in printing them as multiple lines. With this update, exceptions are output as a single line, resolving the issue.(link:https://issues.redhat.com/browse/LOG-3761[LOG-3761])

Reviewer: ```suggestion
* Before this update, the detection of JavaScript client multi-line exceptions by fluentd failed, resulting in printing them as multiple lines. With this update, exceptions are output as a single line, resolving the issue.(link:https://issues.redhat.com/browse/LOG-3761[LOG-3761])
```

Reviewer: same comment re capitalization of fluentd also

> * Before this update, direct upgrades from the `Red Hat Openshift Logging` Operator version 4.6 to version 5.6 were allowed, resulting in functionality issues. With this update, upgrades must be within two versions, resolving the issue. (link:https://issues.redhat.com/browse/LOG-3837[LOG-3837])

Reviewer: ```suggestion
* Before this update, direct upgrades for the Red Hat Openshift Logging Operator from version 4.6 to version 5.6 were allowed, resulting in functionality issues. With this update, upgrades must be within two versions, resolving the issue. (link:https://issues.redhat.com/browse/LOG-3837[LOG-3837])
```
no backticks for names of Operators needed

* Before this update, metrics were not displayed for Splunk or Google Logging outputs. With this update, the issue is resolved by sending metrics for HTTP endpoints.(link:https://issues.redhat.com/browse/LOG-3932[LOG-3932])

* Before this update, when the `ClusterLogForwarder` custom resource was deleted, collector pods remained running. With this update, collector pods do not run when log forwarding is not enabled. (link:https://issues.redhat.com/browse/LOG-4030[LOG-4030])

* Before this update, a time range could not be selected in the {Product-Title} web console by clicking and dragging over the logs histogram. With this update, clicking and dragging can be used to successfully select a time range. (link:https://issues.redhat.com/browse/LOG-4101[LOG-4101])

> * Before this update, fluentd hash values for watch files were generated using the paths to log files, resulting in a non unique hash upon log rotation. With this update, hash values for watch files are created with inode numbers, resolving the issue. (link:https://issues.redhat.com/browse/LOG-3633[LOG-3633])

Reviewer: same fluentd comment

> * Before this update, clicking on the *Show Resources* link in the {Product-Title} web console did not produce any effect. With this update, the issue is resolved by fixing the functionality of the "Show Resources" link to toggle the display of resources for each log entry. (link:https://issues.redhat.com/browse/LOG-4118[LOG-4118])

Reviewer: ```suggestion
* Before this update, clicking on the *Show Resources* link in the {Product-Title} web console did not produce any effect. With this update, the issue is resolved by fixing the functionality of the *Show Resources* link to toggle the display of resources for each log entry. (link:https://issues.redhat.com/browse/LOG-4118[LOG-4118])
```
